### PR TITLE
Ruby: Request headers update.

### PR DESCRIPTION
**Pull request by landrito**
cc @bjwatson

### CODE REVIEW

File: src/main/resources/com/google/api/codegen/ruby/main.snip [hunk 1]
```diff
@@ -228,8 +222,10 @@
         scopes: ALL_SCOPES,
         client_config: {},
         timeout: DEFAULT_TIMEOUT,
-        app_name: "gax",
-        app_version: Google::Gax::VERSION
+        app_name: nil,
+        app_version: nil,
+        lib_name: "gapic",
+        lib_version: "0.1.0"
```
Comment (lukesneeringer) [inline src/main/resources/com/google/api/codegen/ruby/main.snip]: This seems like it is probably wrong (either that or _my_ implementation is wrong, which is totally possible!):

  * I was of the understanding that `lib_name` is supposed to be the library built _on top of_ the GAPIC, not the GAPIC itself. If another library is used, _both_ the `gccl` header _and_ the `gapic` header should be sent, correct?
  * `lib_version` probably should not be hard-coded.

File: src/main/resources/com/google/api/codegen/ruby/main.snip [hunk 2]
```diff
@@ -248,13 +244,20 @@
...
-      google_api_client = "#{app_name}/#{app_version} " @\
-        "#{CODE_GEN_NAME_VERSION} gax/#{Google::Gax::VERSION} " @\
-        "ruby/#{RUBY_VERSION}".freeze
+      if app_name || app_version
+        warn "`app_name` and `app_version` are no longer being used in the request headers."
+      end
+
+      google_api_client = "#{lib_name}/{lib_version} " @\
+        "gax/#{Google::Gax::VERSION} grpc/#{GRPC::VERSION} " @\
+        "gl-ruby/#{RUBY_VERSION}".freeze
```
Comment (lukesneeringer) [inline src/main/resources/com/google/api/codegen/ruby/main.snip]: Move `gl-ruby` to the front of the line (based on changes to the spec from yesterday evening).

File: src/main/resources/com/google/api/codegen/ruby/main.snip (hunk 1 above)
Comment (lukesneeringer) [inline src/main/resources/com/google/api/codegen/ruby/main.snip]: Update: After looking at the spec again, I think I am wrong about the first bullet (the Python implementation is specifically called out as expected to be slightly different from all the others).

Comment (landrito) [inline src/main/resources/com/google/api/codegen/ruby/main.snip]: Agreed that lib_version should not be hard-coded. The prior is hard coded, so lets just hard code for now.

Tracking as an issue: https://github.com/googleapis/toolkit/issues/1015.

File: src/main/resources/com/google/api/codegen/ruby/main.snip (hunk 2 above)
Comment (landrito) [inline src/main/resources/com/google/api/codegen/ruby/main.snip]: Done.

File: src/main/resources/com/google/api/codegen/ruby/main.snip (hunk 1 above)
Comment (lukesneeringer) [inline src/main/resources/com/google/api/codegen/ruby/main.snip]: Update: Based on this morning's changes, this is wrong again. :-)

File: src/main/resources/com/google/api/codegen/ruby/main.snip [hunk 3]
```diff
@@ -248,13 +244,20 @@
...
 
-      google_api_client = "#{app_name}/#{app_version} " @\
-        "#{CODE_GEN_NAME_VERSION} gax/#{Google::Gax::VERSION} " @\
-        "ruby/#{RUBY_VERSION}".freeze
+      if app_name || app_version
+        warn "`app_name` and `app_version` are no longer being used in the request headers."
+      end
+
+      google_api_client = "gl-ruby/#{RUBY_VERSION} " @\
+        "#{lib_name}/{lib_version} gax/#{Google::Gax::VERSION} " @\
```
Comment (lukesneeringer) [inline src/main/resources/com/google/api/codegen/ruby/main.snip]: Add `gapic/#{CODE_GEN_NAME_VERSION}` and make `#{lib_name}/{lib_version}` only be added if `lib_name` is sent.

Comment (lukesneeringer) [inline src/main/resources/com/google/api/codegen/ruby/main.snip]: Also, how are you handling the `gl-ab*` headers?

**Comment by lukesneeringer** [comment]
Looks good except for the lack of A/B.

**Comment by landrito** [comment]
Thanks for the review @lukesneeringer.

**Pull request merged**